app/controllers/course/material/folders_controller.rb

```diff
@@ -3,33 +3,36 @@ class Course::Material::FoldersController < Course::Material::Controller
   before_action :authorize_read_owner!, only: [:show, :download]
 
   def show
-    @subfolders = @folder.children.with_content_statistics.accessible_by(current_ability).
-                  order(:name).includes(:owner).without_empty_linked_folder
-    # Don't display the folder if the user cannot access its owner.
-    @subfolders.select! { |f| can?(:read_owner, f) }
+    respond_to do |format|
+      format.html
+      format.json do
+        @subfolders = @folder.children.with_content_statistics.accessible_by(current_ability).
+                      order(:name).includes(:owner).without_empty_linked_folder
+        # Don't display the folder if the user cannot access its owner.
+        @subfolders.select! { |f| can?(:read_owner, f) }
+      end
+    end
   end
 
   def edit
   end
 
   def update
     if @folder.update(folder_params)
-      redirect_folder = @folder.parent || @folder
-      redirect_to course_material_folder_path(current_course, redirect_folder),
-                  success: t('.success', name: @folder.name)
+      @folder = @folder.parent || @folder
+      show
```

Comment:
Update can be for currFolder or subfolder, and both would require different sets of info. Returning everything was a convenient way to return the correct information since there is no way of telling which variant it is.

Reply:
Gotcha!